### PR TITLE
Add min/max/sum/avg/anyvalue/approxcd agg func

### PR DESCRIPTION
**Pull request by dongdongyang33**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/sql/colexec/agg/anyvalue/anyvalue.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+func (a *Anyvalue[T]) Fill(i int64, value T, ov T, z int64, isEmpty bool, isNull bool) (T, bool) {
+	if !isNull && !a.NotSet[i] {
+		a.NotSet[i] = true
+		return value, false
+	}
+	return ov, isEmpty
+}
+
+func (a *Anyvalue[T]) Merge(xIndex int64, yIndex int64, x T, y T, xEmpty bool, yEmpty bool, yAnyValue any) (T, bool) {
+	if (xEmpty && !yEmpty) || (!xEmpty && !yEmpty) {
```
Comment (fengttt) [inline pkg/sql/colexec/agg/anyvalue/anyvalue.go:40]: Insn't this if test simply if !yEmpty?

File: pkg/sql/colexec/agg/anyvalue/anyvalue.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		ya := yAnyValue.(*Anyvalue[T])
+		if ya.NotSet[yIndex] && !a.NotSet[xIndex] {
+			a.NotSet[xIndex] = true
+			return y, false
```
Comment (fengttt) [inline pkg/sql/colexec/agg/anyvalue/anyvalue.go:44]: This merge code reads very strange.   Will merge add a not set flag?    Very strange for any.

File: pkg/sql/colexec/agg/anyvalue/types.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package anyvalue
+
+import "github.com/matrixorigin/matrixone/pkg/container/types"
+
+type Anyvalue[T any] struct {
+	// the len of NotSet should as same as agg's size
```
Comment (fengttt) [inline pkg/sql/colexec/agg/anyvalue/types.go:20]: What do you mean by "agg's size"?    Number of groups?   Number of elements in one group?   Or the number of bytes of one element in one group?

**Review by fengttt — CHANGES_REQUESTED** [review]
Can you ask MoChen to review, and answer the following question:

Why Fill does not update state, but return a value instead?   Where is the return value stored anyway ...

**Comment by nnsgmsone** [comment]
> 
The return value is stored by the agg framework and the actual aggregation implementation does not need to care how the return value is stored.

File: pkg/sql/colexec/agg/anyvalue/anyvalue.go (hunk 2 above)
Comment (nnsgmsone) [reply pkg/sql/colexec/agg/anyvalue/anyvalue.go:44]: > This merge code reads very strange. Will merge add a not set flag? Very strange for any.

Consider min, then an unset flag may be required, if not set then this value should be set to whatever the minimum is.

File: pkg/sql/colexec/agg/anyvalue/types.go (hunk 3 above)
Comment (nnsgmsone) [reply pkg/sql/colexec/agg/anyvalue/types.go:20]: > What do you mean by "agg's size"? Number of groups? Number of elements in one group? Or the number of bytes of one element in one group?

number of groups

**Comment by nnsgmsone** [comment]
> Can you ask MoChen to review, and answer the following question:
> 
> Why Fill does not update state, but return a value instead? Where is the return value stored anyway ...

The return value is stored by the agg framework and the actual aggregation implementation does not need to care how the return value is stored.

**Comment by nnsgmsone** [comment]
> > Can you ask MoChen to review, and answer the following question:
> > Why Fill does not update state, but return a value instead? Where is the return value stored anyway ...
> 
> The return value is stored by the agg framework and the actual aggregation implementation does not need to care how the return value is stored.

File: pkg/sql/colexec/agg/anyvalue/types.go (hunk 3 above)
Comment (dongdongyang33) [reply pkg/sql/colexec/agg/anyvalue/types.go:20]: Sorry I will remove this annotation. I add it before implementing Grow() function. 
Its size will grow with group size when calling Grow() function.

**Comment by dongdongyang33** [comment]
> > Can you ask MoChen to review, and answer the following question:
> > Why Fill does not update state, but return a value instead? Where is the return value stored anyway ...
> 
> The return value is stored by the agg framework and the actual aggregation implementation does not need to care how the return value is stored.

Thank MoChen reply to the question.
I'm used to add reviewers after passing all BVT/UT tests. I will add reviewers earlier next time

**Review by nnsgmsone — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Pull request merged**